crates/orchestrator/src/jobs/mod.rs

```diff
@@ -173,6 +173,32 @@ pub async fn verify_job(id: Uuid) -> Result<()> {
     Ok(())
 }
 
+/// Terminates the job and updates the status of the job in the DB.
+/// Throws error if the job status `Completed` is existing on DL queue.
+pub async fn handle_job_failure(id: Uuid) -> Result<()> {
+    let config = config().await;
+
+    let mut job = get_job(id).await?.clone();
+    let mut metadata = job.metadata.clone();
+
+    if job.status == JobStatus::Completed {
+        return Err(eyre!("Invalid state exists on DL queue: {}", job.status.to_string()));
```

Comment:
let's log the error and add a TODO to send an alert. we don't really want to retry from SQS here

Reply:
Valid, removed.